subcommand/server-acl-init/rules.go

```diff
@@ -121,6 +127,48 @@ namespace_prefix "" {
 	return c.renderRules(meshGatewayRulesTpl)
 }
 
+func (c *Command) ingressGatewayRules(name, namespace string) (string, error) {
+	ingressGatewayRulesTpl := `
+{{- if .EnableNamespaces }}
+namespace "{{ .GatewayNamespace }}" {
```

Comment:
Given that it's easy to run multiple ingress gateways, I don't think we should support cross-namespace permissions at this point. They're pretty messy anyway and it's better to keep permissions as restricted as possible.

Reply:
I think we should support the use case of ingress gateway in default namespace being able to route to services in other consul namespaces but I think this may already work because of the cross-namespace policy we create in the default namespace.

Can you try it out and let me know if it does?